src/components/MyAccount/CheckoutsTab.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  Box,
+  Button,
+  Icon,
+  Link,
+  Table,
+  Text,
+  useModal,
+} from "@nypl/design-system-react-components"
+import type { Checkout, Patron } from "../../types/accountTypes"
+import { useEffect, useState } from "react"
+import { BASE_URL } from "../../config/constants"
+import { getTitle } from "../../utils/myAccountUtils"
+import styles from "../../../styles/components/MyAccount.module.scss"
+
+const CheckoutsTab = ({
+  checkouts,
+  patron,
+}: {
+  checkouts: Checkout[]
+  patron: Patron
+}) => {
+  const checkoutsHeaders = [
+    "Title",
+    "Barcode",
+    "Call number",
+    "Due back by",
+    "Manage checkout",
+  ]
+
+  function RenewButton(checkout, patron) {
+    const [isButtonDisabled, setButtonDisabled] = useState(false)
+    const { onOpen, Modal } = useModal()
+    const [modalProps, setModalProps] = useState(null)
+    const successModalProps = {
+      bodyContent: (
+        <Box className={styles.modalBody}>
+          <Text sx={{ marginLeft: "l" }}>
+            Your item was renewed. It is now due back on {checkout.date}.
+          </Text>
+        </Box>
+      ),
+      closeButtonLabel: "OK",
+      headingText: (
+        <Box className={styles.modalHeading}>
+          <Icon
+            size="large"
+            name="actionCheckCircleFilled"
+            color="ui.success.primary"
+          />
+          <Text sx={{ marginBottom: "0px" }}> Renewal successful </Text>
+        </Box>
+      ),
+    }
+    const failureModalProps = {
+      bodyContent: (
+        <Box className={styles.modalBody}>
+          <Text sx={{ marginLeft: "l", marginRight: "m" }}>
+            We were unable to renew your item. Please try again or{" "}
+            <Link href="https://www.nypl.org/get-help/contact-us">
+              contact us
+            </Link>{" "}
+            for assistance.
+          </Text>
+        </Box>
+      ),
+      closeButtonLabel: "OK",
+      headingText: (
+        <Box className={styles.modalHeading}>
+          <Icon size="large" name="errorFilled" color="ui.error.primary" />
+          <Text sx={{ marginBottom: "0px" }}> Renewal failed </Text>
+        </Box>
+      ),
+    }
+
+    useEffect(() => {
+      const lastDisabledTime = localStorage.getItem(
+        `lastDisabledTime-${checkout.id}`
+      )
+      if (lastDisabledTime) {
+        const twentyFourHoursAgo = new Date().getTime() - 8640000
+        if (parseInt(lastDisabledTime, 10) > twentyFourHoursAgo) {
+          setButtonDisabled(true)
+        }
+      }
+    }, [])
```

Comment:
If this is meant to only run once on load, why not include it as part of the component logic and not use the `useEffect`?

Reply:
The linter also complains that `checkout.id` is missing from the dep array.

Reply:
It does seem weird but i thought that `localStorage` should be used inside `useEffect` since it's a non react state. Also linter doesn't like it but `checkout.id` isn't really a logical dependency i think